src/components/Pages/Homepage/DocsHeader/DocsHeader.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { InlineSearch } from './InlineSearch';
+import styles from './DocsHeader.module.css';
+
+interface DocsHeaderProps {
+  title?: string;
+  subtitle?: string;
+  hideTitleSection?: boolean;
+  quickActions?: Array<{
+    label: string;
+    href: string;
+  }>;
+}
+
+const defaultQuickActions = [
+  { label: "Enroll Kubernetes cluster", href: "./enroll-resources/kubernetes-access/getting-started/" },
+  { label: "set up SSO with GitHub", href: "./zero-trust-access/sso/github-sso/" },
+  { label: "set up Slack Access Request Plugin", href: "./identity-governance/access-request-plugins/ssh-approval-slack/" }
+];
+
+function DocsHeader({ 
+  title = "Teleport Documentation",
+  hideTitleSection = false,
+  quickActions = defaultQuickActions
+}: DocsHeaderProps) {
+  return (
+    <section className={styles.docsHeader}>
+      {/* Background container with overflow control */}
+      <div className={styles.backgroundContainer}>
+        <div className={styles.background} />
+      </div>
+      
+      {/* Content container without overflow restrictions */}
+      <div className={styles.contentContainer}>
+        <div className={styles.content}>
+          <div className={styles.textSection}>
+            <div className={hideTitleSection ? 'visually-hidden' : styles.title}>{title}</div>
```

Comment:
The layout here feels way more complex than it should be. There's a lot of relatively positioned elements and a lot of divs with a single child, including a flex box with a single item in it. Are we sure all of this is necessary?